### PR TITLE
fnm: init at 1.26.0

### PR DESCRIPTION
**Pull request by kidonng**
###### Motivation for this change

Add [fnm](https://github.com/Schniz/fnm), a Node.js version manager.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kidonng** [comment]
Test failures on Codespaces:

```
failures:
    archive::zip::tests::test_zip_extraction
    commands::install::tests::test_set_default_on_new_installation
    downloader::tests::test_installing_node_12
    downloader::tests::test_installing_npm
    remote_node_index::tests::test_list
    shell::infer::unix::tests::test_get_process_info
```

Wait for ofborg to see if there's something different

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by kidonng** [comment]
Most tests failed due to network problem:

```
Installing Node v8.11.3 (x64)
error: Can't download the requested binary: HttpError: error sending request for url (https://nodejs.org/dist/v8.11.3/node-v8.11.3-linux-x64.tar.xz): error trying to connect: dns error: failed to lookup address information: Name or service not known
```

So I just disabled the test entirely.

File: pkgs/development/tools/fnm/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "Schniz";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-wKtgMUt7QquT6mS3AI+XrZqbJOZxj6jlJi+7uC9w7xU=";
+  };
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  buildInputs = [ ] ++ lib.optionals stdenv.isDarwin [ Security ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fnm/default.nix:16]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [ Security ];
```

File: pkgs/development/tools/fnm/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    installShellCompletion --cmd fnm \
+      --bash <($out/bin/fnm completions --shell bash) \
+      --fish <($out/bin/fnm completions --shell fish) \
+      --zsh <($out/bin/fnm completions --shell zsh)
+  '';
+
+  meta = with lib; {
+    description = "Fast and simple Node.js version manager";
+    homepage = "https://github.com/Schniz/fnm";
+    license = lib.licenses.gpl3Only;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fnm/default.nix:32]:
```suggestion
    license = licenses.gpl3Only;
```

**Pull request merged**